### ISSUE TITLE
COMPAT: handle numpy 1.15 deprecation warnings

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#22101]
I think this is a new warning from numpy in 1.15: https://travis-ci.org/pandas-dev/pandas/jobs/409274058

```
/home/travis/build/pandas-dev/pandas/pandas/core/dtypes/common.py:2026: DeprecationWarning: elementwise == comparison failed; this will raise an error in the future.
    if dtype in [object, np.object_, 'object', 'O']:
```

**Comment by jreback** [comment]
cc @charris

**Comment by charris** [comment]
The `DeprecationWarning` has been around since  NumPy 1.10, what is new is https://github.com/numpy/numpy/pull/10337. Note the Pandas comments in the referenced issue https://github.com/numpy/numpy/issues/5188. I think Pandas may have had a workaround that now fails.

**Comment by jreback** [comment]
closed by #22142
